components/base/wheeled/wheeled_base.go

```diff
@@ -87,13 +93,97 @@ type wheeledBase struct {
 	right     []motor.Motor
 	allMotors []motor.Motor
 
+	mu     sync.Mutex
 	opMgr  operation.SingleOperationManager
 	logger golog.Logger
 
 	name  string
 	frame *referenceframe.LinkConfig
 }
 
+// Reconfigure reconfigures the base atomically and in place.
+func (wb *wheeledBase) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	// resetting the left motor list
+	wb.left = make([]motor.Motor, 0)
+
+	for _, name := range newConf.Left {
+		m, err := motor.FromDependencies(deps, name)
+		if err != nil {
+			return errors.Wrapf(err, "no left motor named (%s)", name)
+		}
+		props, err := m.Properties(ctx, nil)
+		if props[motor.PositionReporting] && err != nil {
+			wb.logger.Debugf("motor %s can report its position for base", name)
+		}
+		wb.left = append(wb.left, m)
+	}
+
+	// resetting the right motor list
+	wb.right = make([]motor.Motor, 0)
+
+	for _, name := range newConf.Right {
+		m, err := motor.FromDependencies(deps, name)
+		if err != nil {
+			return errors.Wrapf(err, "no right motor named (%s)", name)
+		}
+		props, err := m.Properties(ctx, nil)
+		if props[motor.PositionReporting] && err != nil {
+			wb.logger.Debugf("motor %s can report its position for base", name)
+		}
+		wb.right = append(wb.right, m)
+	}
+
+	wb.allMotors = append(wb.allMotors, wb.left...)
+	wb.allMotors = append(wb.allMotors, wb.right...)
+
+	wb.mu.Lock()
+	defer wb.mu.Unlock()
+
+	wb.widthMm = newConf.WidthMM
+	wb.wheelCircumferenceMm = newConf.WheelCircumferenceMM
+	wb.spinSlipFactor = newConf.SpinSlipFactor
+
+	return nil
+}
+
+// CreateWheeledBase returns a new wheeled base defined by the given config.
+func CreateWheeledBase(
```

Comment:
don't move functions around.

Reply:
Is there a reason for this function to be last in this package while other similar functions in other packages are towards the top?

Reply:
no reason - it's just easier to read diffs when the lines are closer. So I'm trying to get everyone in the habit of not moving things around if necessary. 

We can move it around, I just would rather it not be coupled with changes in behaviour. Which is what reconfigure is doing.